core/src/main/java/com/microsoft/applicationinsights/internal/util/LocalStringsUtils.java

```diff
@@ -53,8 +53,15 @@ public static String generateRandomId(boolean removeDashes) {
         return uuid;
     }
 
+    /**
+     * Lazy init pattern.
+     */
+    private static class RandomHolder {
+        static final Random RANDOM = new Random();
```

Comment:
recommend to replace this with ThreadLocalRandom. Has much better performance.

Reply:
Precisely you can use this: https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/ThreadLocalRandom.html#nextLong(long).

For your knowledge : here is why ThreadLocalRandom is better than Random. In nutshell it prevents contention in multithreaded environment where Random synchronizes over the seed.

http://thoughtfuljava.blogspot.com/2012/09/prefer-threadlocalrandom-over-random.html

Reply:
makes sense